swo_aws_extension/management/commands/upload_journal.py

```diff
@@ -0,0 +1,82 @@
+from mpt_extension_sdk.core.utils import setup_client
+from requests.exceptions import HTTPError
+
+from swo_mpt_api import MPTAPIClient
+from swo_mpt_api.models.hints import Journal
+
+
+class Command(BaseCommand):
+    """
+    Upload a journal file to the marketplace
+
+    Upload a file by providing authorization id or journal id.
+
+    Arguments:
+        file: The jsonl file to upload
+        authorization: Optional - The authorization id, it will create
+        journal: Optional - The journal id, it will create
+
+    Example:
+        swoext django upload_journal journal_lines.jsonl --authorization AUT-0001-0001
+
+        swoext django upload_journal journal_lines.jsonl --journal BJO-0005-0005
+    """
+
+    help = "Upload journal file to MPT"
+
+    def success(self, message):
+        self.stdout.write(self.style.SUCCESS(message), ending="\n")
+
+    def info(self, message):
+        self.stdout.write(message, ending="\n")
+
+    def error(self, message):
+        self.stderr.write(self.style.ERROR(message), ending="\n")
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "file", type=argparse.FileType("r"), help="Path to the journal file to upload"
+        )
+
+        group = parser.add_mutually_exclusive_group(required=True)
+        group.add_argument("--authorization", help="Authorization ID to create a new journal")
+        group.add_argument(
+            "--journal", type=str, default=None, help="Existing journal ID to upload to"
+        )
+
+    def handle(self, *args, **options):
+        file = options["file"]
+        authorization = options["authorization"]
+        journal_id = options["journal"]
+
+        client = setup_client()
+        api = MPTAPIClient(client)
+        try:
+            if not journal_id:
+                journal_data = Journal(
+                    authorization={"id": authorization},
+                    name=f"{datetime.now().strftime('%Y-%m-%d')} - Manual upload - {file.name}",
+                )
+                self.info(
+                    f"Creating journal for authorization "
+                    f"{journal_data["authorization"]["id"]} with name `{journal_data["name"]}` ..."
+                )
+                journal = api.billing.journal.create(journal_data)
+                journal_id = journal["id"]
+                self.success(f"Created journal {journal_id} for authorization {authorization}.")
+            self.info(f"Start fileupload {file.name} for authorization {journal_id}...")
+            updated_journal = api.billing.journal.upload(journal_id, file)
+            self.info(
+                f"Uploaded `{file.name}` to journal {journal_id}. "
+                f"Status: `{updated_journal["status"]}` "
+                f"Processing Total: `{updated_journal["processing"]["total"]}`"
+            )
+            self.success(f"File upload {file.name} for authorization {journal_id} completed.")
+        except HTTPError as ex:
```

Comment:
I'd like to add the try/except clause only in the lines where we can expect an exception. This improves the readability and maintainability on the code because you don't need to figure out which lines can raise the exception. On the other hand, it helps avoid a  a mixed bag  of exceptions-> except (HTTPError, IndexError, ....) and a "java" style approach by defining variables first. Pythonic code recommends defining variables close to their use to minimize the need to remember things.